mesonbuild/dependencies/misc.py

```diff
@@ -277,6 +279,211 @@ def need_threads(self):
+                result.append(f)
+                if f in ('-L', '-Xlinker'):
+                    include_next = True
+            elif include_next:
+                include_next = False
+                result.append(f)
+        return result
+
+    def _try_openmpi_wrapper(self, prog):
+        prog = ExternalProgram(prog, silent=True)
+        if prog.found():
+            cmd = prog.get_command() + ['--showme:compile']
+            p, o, e = mesonlib.Popen_safe(cmd)
+            p.wait()
+            if p.returncode != 0:
+                mlog.debug('Command', mlog.bold(cmd), 'failed to run:')
+                mlog.debug(mlog.bold('Standard output\n'), o)
+                mlog.debug(mlog.bold('Standard error\n'), e)
+                return
+            cargs = shlex.split(o)
+
+            cmd = prog.get_command() + ['--showme:link']
+            p, o, e = mesonlib.Popen_safe(cmd)
+            p.wait()
+            if p.returncode != 0:
+                mlog.debug('Command', mlog.bold(cmd), 'failed to run:')
+                mlog.debug(mlog.bold('Standard output\n'), o)
+                mlog.debug(mlog.bold('Standard error\n'), e)
+                return
+            libs = shlex.split(o)
+
+            cmd = prog.get_command() + ['--showme:version']
+            p, o, e = mesonlib.Popen_safe(cmd)
+            p.wait()
+            if p.returncode != 0:
+                mlog.debug('Command', mlog.bold(cmd), 'failed to run:')
+                mlog.debug(mlog.bold('Standard output\n'), o)
+                mlog.debug(mlog.bold('Standard error\n'), e)
+                return
+            version = re.search('\d+.\d+.\d+', o)
+            if version:
+                version = version.group(0)
+            else:
+                version = 'none'
+
+            return version, cargs, libs
+
+    def _try_other_wrapper(self, prog):
+        prog = ExternalProgram(prog, silent=True)
+        if prog.found():
+            cmd = prog.get_command() + ['-show']
+            p, o, e = mesonlib.Popen_safe(cmd)
+            p.wait()
+            if p.returncode != 0:
+                mlog.debug('Command', mlog.bold(cmd), 'failed to run:')
+                mlog.debug(mlog.bold('Standard output\n'), o)
+                mlog.debug(mlog.bold('Standard error\n'), e)
+                return
+            args = shlex.split(o)
+
+            version = 'none'
+
+            return version, args, args
+
+    def _try_msmpi(self):
+        if self.language == 'cpp':
+            # MS-MPI does not support the C++ version of MPI, only the standard C API.
+            return
+        if 'MSMPI_INC' not in os.environ:
+            return
+        incdir = os.environ['MSMPI_INC']
+        arch = detect_cpu_family(self.env.coredata.compilers)
+        if arch == 'x86':
+            if 'MSMPI_LIB32' not in os.environ:
+                return
+            libdir = os.environ['MSMPI_LIB32']
+            post = 'x86'
+        elif arch == 'x86_64':
+            gl = 'lib64*'
```

Comment:
__[flake8]__
* [F841] local variable 'gl' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->